.github/workflows/ci.yml

```
@@ -178,16 +180,77 @@ jobs:
         LLVM_PROFILE_FILE="coverage-%p.profraw" ctest --test-dir build --output-on-failure
         llvm-profdata-18 merge -sparse coverage-*.profraw -o coverage.profdata
         llvm-cov-18 report ./build/bin/* -instr-profile=coverage.profdata
+        # Export LCOV for Codecov
+        llvm-cov-18 export ./build/bin/* -instr-profile=coverage.profdata -format=lcov > coverage.lcov
 
     - name: Upload coverage reports
       uses: codecov/codecov-action@v3
```

actionlint (1.7.8)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)